### PR TITLE
QoL improvements for Makefile and .gitignore

### PR DESCRIPTION
**Pull request by davidz627**
Updated Makefile so that it doesn't recompile stale binaries. Added gitignore entries for common items and binaries

### CODE REVIEW

**Comment by saad-ali** [comment]
/lgtm

@jsafrane for review and approval

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# Compiled binaries and deployment files 
+/bin/
+/deploy/
```
Comment (jsafrane) [inline .gitignore]: I'd like to ignore csi-attacher binary and other compiled artifacts explicitly in `deploy/`. The directory can have useful files that should not be ignored.

File: .gitignore [hunk 2]
```diff
@@ -12,3 +16,38 @@
...
+.settings/**
+
+# Files generated by JetBrains IDEs, e.g. IntelliJ IDEA
+.idea/
+*.iml
+
+# Vscode files
+.vscode
+
+# This is where the result of the go build goes
```
Comment (jsafrane) [inline .gitignore]: `go build` does not create anything in `*output*/`

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,7 @@
+# Compiled binaries and deployment files 
+/bin/
+/deploy/
+
 # Binaries for programs and plugins
```
Comment (jsafrane) [inline .gitignore]: I think you can prune existing .gitignore too, e.g. we don't use glide.

**Comment by davidz627** [comment]
@jsafrane comments addressed. PTAL

**Pull request merged**